gammapy/cube/basic_cube.py

```diff
@@ -0,0 +1,62 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Functions to perform basic functions for map and cube analysis.
+"""
+from __future__ import absolute_import, division, print_function, unicode_literals
+from ..maps import WcsNDMap
+
+def make_map_counts(evts, valid_ref_map):
+    """
+    Build a ``WcsNDMap` with events from an EventList.
+    The energy of the events is used for a non-spatial axis homogeneous to energy.
+    The other non-spatial axis names should have an entry in the colum names of the ``EventList``
+
+    Parameters
+    __________
+
+    evts : `~gammapy.data.EventList`
+            the input event list
+    valid_ref_map : `~gammapy.maps.WcsNDMap`
+        Map containing the valid pixels
+
+    Returns
+    -------
+    cntmap : `~gammapy.maps.WcsNDMap`
+        Count cube (3D) in true energy bins
+
+    """
+    # The list will contain the event table entries to be fed into the WcsNDMap
+    tmp = list()
+    ref_geom = valid_ref_map.geom
+    # Convert events coordinates
+    if ref_geom.coordsys == 'GAL':
+        tmp.append(evts.galactic.l)
```

Comment:
This is ugly. I'm not sure yet how to do the frame handling in the end. But for now, at least avoid doing `events. galactic` two times. I.e. compute the coord in the right frame once, and then after the if / else, access lon / lat via `coord.data.lon` and `coord.data.lat` (I think those are the names, not 100% sure).